internal/debug/debug.go

```diff
@@ -0,0 +1,29 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package debug // import "go.opentelemetry.io/otel/internal/debug"
+
+import (
+	"github.com/go-logr/logr"
+)
+
+var Log logr.Logger = logr.Discard()
```

Comment:
Can we unexport this global variable and add a `SetLogger` function to be able to set it?

This way we can make sure that in other packages we can use only functions exposed by this package instead of everything that `logr` exposes (e.g. use some magic logging level like in `sdk/trace/batch_span_processor.go`😄 )

Reply:
I think that exposing the V() and WithValues() methods internally are the compelling features of this approach.

Reply:
Unexported, and exposed a smaller API